lib/base/index.js

```diff
@@ -1,136 +1,138 @@
 // @flow
 import {keybaseExec, rmdirRecursive, randomTempDir, keybaseServiceStartup} from '../utils'
-import {withStore, actionInit, actionPreinit} from '../store'
-import type {Store} from '../store/types'
-import type {DeviceUsernamePair} from '../types'
-import type {Init, Deinit} from './types'
 
-/**
- * @ignore
- * getCurrentUsernameAndDevicename returns { username, devicename } from keybase status -j
- */
-const getCurrentUsernameAndDevicename = async (store: Store): Promise<DeviceUsernamePair> => {
-  const {homeDir} = store.getState()
-  const status = await keybaseExec(homeDir, ['status', '--json'], {
-    json: true,
-  })
-  if (status && status.Username && status.Device && status.Device.name) {
-    return {username: status.Username, devicename: status.Device.name}
-  } else {
-    throw new Error('failed to get username + device name')
-  }
-}
+export type Initialized = false | 'paperkey' | 'currentService'
 
-/**
- * @ignore
- * guardInitialized takes the current store and the api function to guard
- * It ensures that the following conditions remain true
- *  1. The bot was initialized before calling the given api function
- *  2. The bot has username and devicename if it has been initialized
- *  2. The current username and devicename have not changed values
- *
- * If any condition is false, an error is thrown
- * Otherwise, return the function (to be executed later)
- *
- */
-export const guardInitialized = async (store: Store): Promise<void> => {
-  const currentDPair = await getCurrentUsernameAndDevicename(store)
-  const {initialized, username, devicename} = store.getState()
-  if (
-    !initialized ||
-    !currentDPair ||
-    !username ||
-    !devicename ||
-    currentDPair.username !== username ||
-    currentDPair.devicename !== devicename
-  ) {
-    throw new Error('Uh-oh, username has changed or we never initialized correctly.')
-  }
-}
+export type InitOptions = {|
+  verbose?: boolean,
+  homeDir?: string,
```

Comment:
I think it makes sense to allow users to define their own `homeDir` if they so desire, and randomly generate one if not specified.

Reply:
Allowing this may be dangerous with the current code, as the cleanup process expects this to have been generated randomly and have a big random string in it. Deinit right now (1) deletes the directory, and (2) kills any process that _mentions_ the directory. (This second technique isn't totally ideal; I'd like to change it to keep track of spawned pids.) 

In other words, imagine someone passed a homeDir that had shit in it, because they didn't really understand the argument. And then it nuked their files.

So, if you allow passing one, you'll need to make 2 changes:
(1) confirming the directory doesn't already exist (OR if it does only delete known Keybase files/DB's)
(2) change the killing of the listen processes to be smarter than a regex for the homedir.

I guess for now I'd propose dropping this and creating a separate issue for allowing people to specify a home directory.

Reply:
Good point. We do need a way right now to give sub bot the current directory of its parent unless we want a bunch of different services running for just one bot, but I'll make that into a private method for now that we can make public in a future PR.